ocean_py/asset/asset_off_chain.py

```diff
@@ -0,0 +1,81 @@
+"""
+    Asset class to hold Ocean asset information such as asset id and metadata
+"""
+import re
+from web3 import Web3
+
+from squid_py.did import (
+    did_parse,
+    id_to_did,
+)
+
+from ocean_py.asset.asset_base import AssetBase
+# from ocean_py import logger
+
+class AssetOffChain(AssetBase):
+    def __init__(self, ocean, did=None, agent=None, **kwargs):
+        """
+        init an asset class with the following:
+        :param ocean: ocean object to use to connect to the ocean network
+        :param did: Optional did of the asset
+        """
```

Comment:
What does it mean (i.e. what is the use case) for when an asset is constructed without a DID?

Reply:
If it's a new asset, then we do not know the DID of the asset. So the asset object is created with no assigned DID. In this case it's an off chain asset, so the DID format will be 'op:did:<did-of-agent>/<asset_id>'